### PR TITLE
Step 3 - Tripal Importer - Form implementation

### PR DESCRIPTION
**Pull request by risharde**
Another basic PR with a file addition. This is the Tripal Importer Form (UI) which the future Importers will use (upcoming PR will include the importer include files)

@laceysanderson

### CODE REVIEW

File: tripal/src/Form/TripalImporterForm.php [hunk 1]
```diff
@@ -0,0 +1,570 @@
...
+      // not add a job.
+      if ($form_state->isRebuilding() == TRUE) {
+        return;
+      }
+
+      $importer->create($run_args, $file_details);
+      $importer->submitJob();
+
+    } catch (Exception $e) {
+        drupal_set_message('Cannot submit import: ' . $e->getMessage(), 'error');
```
Comment (laceysanderson) [inline tripal/src/Form/TripalImporterForm.php]: drupal_set_message() is deprecated.

File: tripal/src/Form/TripalImporterForm.php [hunk 2]
```diff
@@ -0,0 +1,570 @@
...
+  }  
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+
+
+    // TODO: Convert the below D7 code to D8/D9
+    global $user;
```
Comment (laceysanderson) [inline tripal/src/Form/TripalImporterForm.php]: Deprecated: https://www.drupal.org/node/2032447

File: tripal/src/Form/TripalImporterForm.php [hunk 3]
```diff
@@ -0,0 +1,570 @@
...
+    } catch (Exception $e) {
+        drupal_set_message('Cannot submit import: ' . $e->getMessage(), 'error');
+    }  
+  }
+
+    /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    // Convert the validation code into the D8/9 equivalent
```
Comment (laceysanderson) [inline tripal/src/Form/TripalImporterForm.php:268]: Is this comment no longer correct? It looks like the code has been updated...

File: tripal/src/Form/TripalImporterForm.php [hunk 4]
```diff
@@ -0,0 +1,570 @@
...
+    $importer = new $class();
+    $importer->formValidate($form, $form_state);    
+
+
+  }
+
+  /**
+   * Build the form for a TripalImporter implementation.
+   */
+  function tripal_get_importer_form($form, &$form_state, $class) {
```
Comment (laceysanderson) [inline tripal/src/Form/TripalImporterForm.php]: Are these the old D7 versions? If so, they should be removed entirely at this point.

File: tripal/src/Form/TripalImporterForm.php [hunk 5]
```diff
@@ -0,0 +1,570 @@
...
+      '#value' => t($class::$button_text),
+      '#weight' => 10,
+    ];
+    return $form;
+  }
+
+  /**
+   * Validate function for the tripal_get_importer_form form().
+   */
+  function tripal_get_importer_form_validate($form, &$form_state) {
```
Comment (laceysanderson) [inline tripal/src/Form/TripalImporterForm.php]: Are these the old D7 versions? If so, they should be removed entirely at this point.

File: tripal/src/Form/TripalImporterForm.php [hunk 6]
```diff
@@ -0,0 +1,570 @@
...
+
+    // Now allow the loader to do validation of it's form additions.
+    $importer = new $class();
+    $importer->formValidate($form, $form_state);
+  }
+
+  /**
+   * Submit function for the tripal_get_importer_form form().
+   */
+  function tripal_get_importer_form_submit($form, &$form_state) {
```
Comment (laceysanderson) [inline tripal/src/Form/TripalImporterForm.php]: Are these the old D7 versions? If so, they should be removed entirely at this point.

**Review by laceysanderson — CHANGES_REQUESTED** [review]

**Comment by risharde** [comment]
@laceysanderson thanks for reviewing, indeed you were correct - I used the old functions to create the new ones and didn't remove them. I've double checked that these functions don't seem to be in use on the form side of things so they've now been removed. Also deprecated / incompatible code removed as requested.

**Review by laceysanderson — APPROVED** [review]

**Pull request merged**